### PR TITLE
MXKRoomDataSource: Fix missing local echos

### PR DESCRIPTION
**Pull request by ismailgulek**
Fix vector-im/element-ios#1771

### CODE REVIEW

File: MatrixKit/Models/Room/MXKRoomDataSource.m [hunk 1]
```diff
@@ -3054,6 +3040,15 @@ - (void)processQueuedEvents:(void (^)(NSUInteger addedHistoryCellNb, NSUInteger
 
                 for (MXKQueuedEvent *queuedEvent in self->eventsToProcessSnapshot)
                 {
+                    @synchronized (self->eventIdToBubbleMap)
+                    {
+                        //  Check whether the event processed before
+                        if (self->eventIdToBubbleMap[queuedEvent.event.eventId])
+                        {
+                            continue;
```
Comment (manuroe) [inline MatrixKit/Models/Room/MXKRoomDataSource.m:3048]: Can we log the event id and its state?

File: MatrixKit/Models/Room/MXKRoomDataSource.m [hunk 2]
```diff
@@ -2283,27 +2283,13 @@ - (void)handleUnsentMessages
     NSArray<MXEvent*>* outgoingMessages = _room.outgoingMessages;
     
     [self.mxSession decryptEvents:outgoingMessages inTimeline:nil onComplete:^(NSArray<MXEvent *> *failedEvents) {
-        BOOL shouldProcessQueuedEvents = NO;
         
-        for (NSInteger index = 0; index < outgoingMessages.count; index++)
+        for (MXEvent *outgoingMessage in outgoingMessages)
```
Comment (manuroe) [inline MatrixKit/Models/Room/MXKRoomDataSource.m:2287]: Can we log the number of outgoing events we push?

**Review by manuroe — COMMENTED** [review]
It looks promising. More logs may be useful for further debugging.

File: MatrixKit/Models/Room/MXKRoomDataSource.m (hunk 1 above)
Comment (ismailgulek) [reply MatrixKit/Models/Room/MXKRoomDataSource.m:3051]: Added

File: MatrixKit/Models/Room/MXKRoomDataSource.m (hunk 2 above)
Comment (ismailgulek) [reply MatrixKit/Models/Room/MXKRoomDataSource.m:2287]: Added

**Review by manuroe — APPROVED** [review]

**Pull request merged**